src/Ledger/Utxo.lagda

```diff
@@ -335,22 +336,66 @@ pattern UTXO-inductive⋯ tx Γ s x y z w k l m n o p q r
       = UTXO-inductive {tx}{Γ}{s} (x , y , z , w , k , l , m , n , o , p , q , r)
 unquoteDecl UTXO-premises = genPremises UTXO-premises (quote UTXO-inductive)
 
+dm : ∀{P Q : Set} → ⦃ P ⁇ ⦄ → ⦃ Q ⁇ ⦄ → ¬ (P × Q) → ¬ P ⊎ ¬ Q
+dm ⦃ ⁇ no ¬p ⦄ ⦃ _ ⦄ ¬pq = inj₁ ¬p
+dm ⦃ _ ⦄ ⦃ ⁇ no ¬q ⦄ ¬pq = inj₂ ¬q
+dm ⦃ ⁇ yes p ⦄ ⦃ ⁇ yes q ⦄ ¬pq = contradiction (p , q) ¬pq
+
 instance
-  Computational-UTXO : Computational _⊢_⇀⦇_,UTXO⦈_
+  Computational-UTXO : Computational _⊢_⇀⦇_,UTXO⦈_ String
```

Comment:
I think now would be a good time to move this into the `Ledger.Utxo.Properties` module